android/src/com/mapswithme/maps/settings/StoragePathManager.java

```diff
@@ -454,7 +442,7 @@ private static int changeStorage(StorageItem newStorage, StorageItem oldStorage)
         throw new IllegalStateException("Cannot move maps. New path is not a directory. New path : " + newDir);
       if (!oldDir.isDirectory())
         throw new IllegalStateException("Cannot move maps. Old path is not a directory. Old path : " + oldDir);
-      if (!StorageUtils.isDirWritable(fullNewPath))
```

Comment:
Our check is better that `File.canWrite()`!

Reply:
The isDirWritable() check uses mkdir() and hence fails on volumes with 0 free space. But we need to show them to the user, especially if this volume is currently configured one.
We've discussed it elsewhere in the comments already.

Reply:
@pastk Can the user select a destination with 0 free space to initiate moving the maps? What will happen?